lib/hocon/impl/config_delayed_merge.rb

```diff
@@ -56,4 +56,79 @@ def hash
     # note that "origin" is deliberately NOT part of equality
     @stack.hash
   end
+
+  def render_to_sb(sb, indent, at_root, at_key, options)
```

Comment:
So, I may just be confused about the ways we've had to rename the various render methods to deal with all of the signature overloads, but I was thinking that the one that you need to override in subclasses of AbstractConfigObject was `render_value_to_sb`.  If this is working for you then I probably just have some assumption wrong somewhere, but do you know where this gets called from?  (I'll poke through the code to look for that in a sec, after I've reviewed the rest of the PR.)